### PR TITLE
potential fix for incorrect gap junction numbers

### PR DESCRIPTION
**Pull request by d2fong**
Should not be merged yet still need to test and add automated test.

refs #67

### CODE REVIEW

File: src/server/db/nematode-connections.js [hunk 1]
```diff
@@ -136,7 +136,46 @@ let queryNematodeConnections = async (connection, opts) => {
...
   });
 
-  return connections;
+  // gap junctions are unordered but sometimes we have entries for both pairs
+  // e.g. 'AIA-ASI' has entries and 'ASI-AIA' has entries
+  // merge them before returning them
+
+  const gapJunctions = connections.filter(c => c.type == 'electrical');
+  const chemicalSynapses = connections.filter(c => c.type == 'chemical');
+
```
Comment (d2fong) [inline src/server/db/nematode-connections.js:145]: Partition connections into gap junctions and chemical synapses

File: src/server/db/nematode-connections.js [hunk 2]
```diff
@@ -136,7 +136,46 @@ let queryNematodeConnections = async (connection, opts) => {
...
+  const gapJunctionsKeyMap = {};
+
+  gapJunctions.forEach(gj => {
+    const { pre, post, synapses } = gj;
+    const prePost = [pre, post];
+    prePost.sort();
+    const key = prePost.join('$');
```
Comment (d2fong) [inline src/server/db/nematode-connections.js:152]: For each gap junction create a key such that partner order is not a factor by sorting pre/post alphabetically.
e.g. 
'AIAR ASIR' maps to 'AIAR$ASIR'
'ASIR AIAR' maps to 'AIAR$ASIR'

File: src/server/db/nematode-connections.js [hunk 3]
```diff
@@ -136,7 +136,46 @@ let queryNematodeConnections = async (connection, opts) => {
...
+
+    if(gapJunctionsKeyMap[key] == null){
+      gapJunctionsKeyMap[key] = gj;
+    } else {
+      Object.keys(synapses).forEach(dataset => {
+        if(gapJunctionsKeyMap[key]['synapses'][dataset] != null){
+          gapJunctionsKeyMap[key]['synapses'][dataset] += synapses[dataset];
+        } else {
+          gapJunctionsKeyMap[key]['synapses'][dataset] = synapses[dataset];
+        }
```
Comment (d2fong) [inline src/server/db/nematode-connections.js:162]: 157-162 merges gap junction counts with the same keys.

File: src/server/db/nematode-connections.js [hunk 4]
```diff
@@ -136,7 +136,46 @@ let queryNematodeConnections = async (connection, opts) => {
...
+
+    return {
+      pre,
+      post,
+      type: gjData.type,
+      synapses: gjData.synapses,
+      annotations: gjData.annotations
+    }
+  });
+  return [...mergedGapJunctions, ...chemicalSynapses];
```
Comment (d2fong) [inline src/server/db/nematode-connections.js:178]: Don't change the connections that are chemical synapses, but merge the gap junctions

**Review by dwitvliet — APPROVED** [review]
Awesome, that looks good to me!

**Comment by d2fong** [comment]
I think the tests are passing but the lint check fails.

**Comment by dwitvliet** [comment]
Is should be good now?

**Comment by d2fong** [comment]
Oh right.  Is there a command for that?

**Comment by dwitvliet** [comment]
> Oh right. Is there a command for that?

Do you mean `npm run test:eslint`?

**Comment by d2fong** [comment]
Does that auto fix the lint issues?  I thought there was one that did.

**Comment by d2fong** [comment]
Yeah it should be.  Let's just merge it.

**Pull request merged**